.circleci/config.yml

```diff
@@ -75,106 +66,327 @@ commands:
-          study_guid: ANGIO
+      - app-build-and-store:
+          study_key: << pipeline.parameters.study_key >>
+
+  osteo-deploy-only:
+    executor:
+      name: commit-executor
+    steps:
+      - deploy-stored-build: &osteo-params
+          study_key: osteo
+          study_guid: CMI-OSTEO
 
-  brain:
+  brain-deploy-only:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
+      - deploy-stored-build: &brain-params
           study_key: brain
           study_guid: cmi-brain
 
-  mbc:
+  angio-deploy-only:
+    executor:
+      name: commit-executor
+    steps:
+      - deploy-stored-build: &angio-params
+          study_key: angio
+          study_guid: ANGIO
+
+  mbc-deploy-only:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
+      - deploy-stored-build: &mbc-params
           study_key: mbc
           study_guid: cmi-mbc
 
+  osteo-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *osteo-params
+
+  angio-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *angio-params
+
+  brain-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *brain-params
+
+  mbc-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *mbc-params
+
+parameters:
+  study_key:
+    type: string
+    default: "UNKNOWN"
+  do-builds:
+    type: boolean
+    default: false
+
 workflows:
   version: 2
-  build-deploy:
+
+  build-and-deploy-all-apps-workflow:
     jobs:
-      - osteo:
-          # todo: more DRY on branch filtering
-          filters:
-            branches:
-              only:
-                - develop
-      - angio:
+      - osteo-build-and-deploy: &build-and-deploy-filters
           filters:
             branches:
               only:
                 - develop
-      - brain:
+      - brain-build-and-deploy:
+          <<: *build-and-deploy-filters
+      - angio-build-and-deploy:
+          <<: *build-and-deploy-filters
+      - mbc-build-and-deploy:
+          <<: *build-and-deploy-filters
+
+  launch-all-app-builds-workflow:
+    # using do-builds param to select which of the two workflows using the build-on-tag-filters
+    # should run; should be this one that just launches the builds or the one actually doing the builds
+    unless: << pipeline.parameters.do-builds >>
+    jobs:
+      - conditionally-launch-build-and-store-all-job: &build-and-store-filters
           filters:
             branches:
               only:
-                - develop
-      - mbc:
+                - /^rc.*/
+                - /^hotfix.*/
+
+  build-app-workflow:
+    when: << pipeline.parameters.do-builds >>
+    jobs:
+      - app-build-and-store-job:
+          <<: *build-and-store-filters
+
```

Comment:
The two workflows above work together and off the same branches; the first one triggers the second one.